module/metrics/hotstuff.go

```diff
@@ -277,3 +278,11 @@ func (hc *HotstuffCollector) ValidatorProcessingDuration(duration time.Duration)
 func (hc *HotstuffCollector) PayloadProductionDuration(duration time.Duration) {
 	hc.payloadProductionDuration.Observe(duration.Seconds()) // unit: seconds; with float64 precision
 }
+
+// TimeoutCollectorsRange reports information about state of timeout collectors, it measurers how many
+// timeout collectors were created and what is the lowest/highest retained view for them.
+func (hc *HotstuffCollector) TimeoutCollectorsRange(lowestRetainedView uint64, newestViewCreatedCollector uint64, activeCollectors int) {
+	hc.timeoutCollectorsRange.WithLabelValues("lowest_retained_view").Set(float64(lowestRetainedView))
+	hc.timeoutCollectorsRange.WithLabelValues("newest_view_of_created_collector").Set(float64(newestViewCreatedCollector))
+	hc.timeoutCollectorsRange.WithLabelValues("active_collectors").Set(float64(activeCollectors))
```

Comment:
Prometheus labels should be used to differentiate different measurements of the same metric (they should measure the same thing, in the same units). Here we're combining two (arguably 3) different measurements into the same metric. For example `lowest_retained_view` is measuring a view number where `active_collectors` is measuring a count.

Suggestion: Create a separate metric for `n_ active_collectors`.

Reply:
It's functionally correct, but I think it's going against best practices, because the gauge vector is combining conceptually different metrics (concretely: metrics with different base units) into the same vector. Here is the guidelines from Prometheus' documentation:

> As a rule of thumb, either the sum() or the avg() over all dimensions of a given metric should be meaningful (though not necessarily useful). If it is not meaningful, split the data up into multiple metrics. For example, having the capacity of various queues in one metric is good, while mixing the capacity of a queue with the current number of elements in the queue is not.

Here, aggregating the metrics with unit "view" and the metric with unit "number of collectors" isn't meaningful - hence why I think it should be a separate metric.

Reply:
https://github.com/onflow/flow-go/pull/4311/commits/7446b996ed60c6137bccfdb72cce0e31a5a8d2ef